source/shared/cpp/ObjectModel/ParseUtil.cpp

```diff
@@ -334,6 +337,23 @@ std::vector<std::shared_ptr<BaseCardElement>> ParseUtil::GetElementCollection(
         {
             // Use the parser that maps to the type
             elements.push_back(parser->Deserialize(elementParserRegistration, actionParserRegistration, curJsonValue));
+            if (elements.back()->GetElementType() == CardElementType::TextBlock)
+            {
+                auto textBlock = std::dynamic_pointer_cast<TextBlock>(elements.back()); 
```

Comment:
i learned we can't use dynamic cast for codes bound for windows integration

Reply:
there are multiple uses of dynamic_pointer_cast in AdaptiveColumn, AdaptiveContainer, Util.cpp, etc. should we open an issue to fix those casts too?

Reply:
if they were there before, dynamic_pointer_cast is probably o.k. to use. i meant it for dynamic_cast, you probably want to check it with Becky if you wanted to be sure.

---
In reply to: [166155402](https://github.com/Microsoft/AdaptiveCards/pull/1102#discussion_r166155402) [](ancestors = 166155402)

Reply:
I believe we can't use dyanmic_pointer_cast in windows.  There are places where i used them before, but when Ricardo ported this code to windows he removed them from the windows version. As of right now i believe that the windows version of the code is different than the github version of the code, which is why you still see them in github.

---
In reply to: [166159978](https://github.com/Microsoft/AdaptiveCards/pull/1102#discussion_r166159978) [](ancestors = 166159978,166155402)